src/ensRegistry.ts

```diff
@@ -1,73 +1,93 @@
 // Import types and APIs from graph-ts
 import {
-  Address,
-  Bytes,
   ByteArray,
   crypto,
+  ens,
 } from '@graphprotocol/graph-ts'
 
 // Import event types from the registry contract ABI
-import { NewOwner, Transfer as TransferEvent, NewResolver, NewTTL } from './types/ENSRegistry/EnsRegistry'
+import { NewOwner, Transfer, NewResolver, NewTTL } from './types/ENSRegistry/EnsRegistry'
 
 // Import entity types generated from the GraphQL schema
-import { Domain, Account } from './types/schema'
+import { Account, Domain, Resolver } from './types/schema'
 
 // Handler for NewOwner events
-export function newOwner(event: NewOwner): void {
-  let subnode = crypto.keccak256(concat(event.params.node, event.params.label)).toHex()
-
-  let account = new Account(event.params.owner.toHex())
+export function handleNewOwner(event: NewOwner): void {
+  let account = Account.load(event.params.owner.toHexString())
+  if (account == null){
+    account = new Account(event.params.owner.toHexString())
+    account.domainCount = 0
+  }
+  account.domainCount = account.domainCount + 1
   account.save()
 
+  let subnode = crypto.keccak256(concat(event.params.node, event.params.label)).toHexString()
   let domain = new Domain(subnode)
+
+  // Get label and node names
+  let label = ens.nameByHash(event.params.label.toHexString())
+  if (label != null) {
+    domain.labelName = label
+  }
+
   domain.owner = account.id
-  domain.parent = event.params.node.toHex()
+  domain.parent = event.params.node.toHexString()
   domain.labelhash = event.params.label
   domain.save()
-
 }
 
 // Handler for Transfer events
-export function transfer(event: TransferEvent): void {
-  let node = event.params.node.toHex()
+export function handleTransfer(event: Transfer): void {
+  let node = event.params.node.toHexString()
+  let oldDomain = Domain.load(node)
 
-  // // Create transfer if it does not exists yet
-  // let transfer = store.get('Transfer', domainId) as Transfer | null
-  // if (transfer == null) {
-  //   transfer = new Transfer()
-  //   transfer.domain = domainId
-  //   transfer.owners = new Array<Bytes>()
-  // }
-  //
-  // // Add the new owner to the list of historical owners of the domain
-  // let owners = transfer.owners
-  // owners.push(event.params.owner)
-  // transfer.owners = owners
+  // if domain does exist, we must minus that owners domain count
+  if (oldDomain != null) {
+    let oldOwner = Account.load(oldDomain.owner)
+    if (oldOwner == null){
```

Comment:
Shouldn't this be impossible?

Reply:
Yes, it should be, I have just been burned many times with subgraphs, where the smart contract code is complex , and then after 6 or 7 million blocks an edge case appears where the entity wasn't created and then the subgraph fails. I've started to do this more often than not for making the subgraph more reliable. Because sometimes the failures even happen when the subgraph is live, and then we have to quickly redeploy.

However, this part of ENS is pretty straight forward and it seems pretty clear that it shouldn't crash. So I will remove it, and then double check my having a local subgraph fully sync with no failures.